optimism/src/mips/interpreter.rs

```diff
@@ -127,18 +127,35 @@ pub struct Signed<T> {
     pub magnitude: T,
 }
 
+#[derive(Copy, Clone)]
+pub enum LookupMode {
+    Read,
+    Write,
+}
+
 impl<T: One> Signed<T> {
-    pub fn read_one() -> Self {
-        Self {
-            sign: Sign::Neg,
-            magnitude: T::one(),
+    /// Creates a new Signed element, either a Read or a Write, and can be null if the flag is zero
+    // TODO: if the flag trick works, then RC does not need to be length 25 anymore nor nonzero default column values
```

Comment:
Can you explain the 'flag trick' here? The magnitude should only be used when we use the final tables, to account for the number of times we lookup a value.
For example, if we lookup values
```
[ 0, 0, 6, 6, 0, 6, 0 ]
```
in table
```
[ 0, 1, 2, 3, 4, 5, 6, 7 ]
```
then we would expect the table magnitudes to be
```
[ 4, 0, 0, 0, 0, 0, 3, 0 ]
```
so that all of the `1`s from the values cancel out with the table's magnitudes.

Reply:
Oh, never mind, I see that it's either `0` or `1`

Reply:
I was expecting the magnitude to represent something like _the number of times this value is added to the table_ . In particular, when I use them inside Keccak, the magnitude that is inferred from flags is either 1 or 0. Meaning, it either adds the lookup (as either positive or negative), or it does not because the sign is multiplied by zero. I had understood that was how magnitude was being used after reading the code of the MIPs interpreter where magnitude is sometimes `if_is_true`.

Reply:
Meaning, the lookups are there, it doesn't mean that different rows have a different set of lookups. It's just when variables are actual witness values that those terms in the lists would get nullified.